### DISCUSSION TITLE
Null component is compiled

### DISCUSSION THREAD

**Discussion by ToniJobiqo** [vercel/next.js#34090] [Help]
In the company, we're trying to reduce app size and we noticed that components returning null conditionally get compiled when running  `> next build` only if they are used within server-side rendered pages.
But if they are rendered within a statically rendered page they are not compiled.

An **important** thing to note is that at the moment of building condition is `false`.

1. If at the moment of building condition is `false`, shouldn't we expect that the component is not compiled?
2. Is this behaviour expected? 
3. How can we be sure that the component returning null at build time will not be compiled?
4. Is the getInitialProps related to this behaviour?

Here is an example of a [component](https://github.com/ToniJobiqo/feature-flag-test/blob/main/components/TestDiv.tsx
) that returns null conditionally.

We are trying to implement a feature flags system, and when the flag is falsy and we return null from the component we do not want it to be compiled, any suggestions on how to approach this?

**Comment by joaogarin** [comment]
This discussion seems somewhat similar? https://github.com/vercel/next.js/discussions/34058

But not sure its aiming at the same point

**Reply by joaogarin** [reply]
May be worth checking using env variables that are prefixed with `NEXT_PUBLIC` 

https://nextjs.org/docs/basic-features/environment-variables#exposing-environment-variables-to-the-browser

**Reply by ToniJobiqo** [reply]
I've already tried with `NEXT_PUBLIC`, thanks for the answer. Even if the prefixed env variable is false and the component returns null it is still compiled. That would be a possible approach to feature flags, but there is still an issue that component is compiled despite it returning null, and also I'd like to know the answer on the 4th.

> 4. Is the getInitialProps related to this behaviour?
